tests/list1.py

```diff
@@ -20,3 +20,4 @@ def test_List():
     a.append(10)
     a.remove(1)
     a.insert(2, 13)
+    a += [4, 5]
```

Comment:
Let's also add a test like this to make sure it works too:
```suggestion
    a += [4, 5]
    a = [6, 7] + a
```

Reply:
Don't forget about this test.

Reply:
We need to improve `check_equal_type` in ASR utils to make this test work.

Reply:
I realized that the existing test worked because currently the `augassign` visitor doesn't really check the types. We should fix that as well.

Reply:
Done!